sycl/source/detail/scheduler/commands.cpp

```diff
@@ -844,62 +525,17 @@ Command *Command::addDep(DepDesc NewDep, std::vector<Command *> &ToCleanUp) {
       NewDep.MDepCommand->addUser(this);
   }
 
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  emitEdgeEventForCommandDependence(NewDep.MDepCommand,
-                                    (void *)NewDep.MDepRequirement->MSYCLMemObj,
-                                    true, NewDep.MDepRequirement->MAccessMode);
-#endif
-
   return ConnectionCmd;
 }
 
 Command *Command::addDep(EventImplPtr Event,
                          std::vector<Command *> &ToCleanUp) {
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  // We need this for just the instrumentation, so guarding it will prevent
-  // unused variable warnings when instrumentation is turned off
-  Command *Cmd = (Command *)Event->getCommand();
-  ur_event_handle_t UrEventAddr = Event->getHandle();
-  // Now make an edge for the dependent event
-  emitEdgeEventForEventDependence(Cmd, UrEventAddr);
-#endif
-
   return processDepEvent(std::move(Event), DepDesc{nullptr, nullptr, nullptr},
                          ToCleanUp);
 }
 
-void Command::emitEnqueuedEventSignal(const ur_event_handle_t UrEventAddr) {
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  emitInstrumentationGeneral(
-      MStreamID, MInstanceID, static_cast<xpti_td *>(MTraceEvent),
-      xpti::trace_signal, static_cast<const void *>(UrEventAddr));
-#endif
-  std::ignore = UrEventAddr;
-}
-
-void Command::emitInstrumentation(uint16_t Type, const char *Txt) {
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  return emitInstrumentationGeneral(MStreamID, MInstanceID,
-                                    static_cast<xpti_td *>(MTraceEvent), Type,
-                                    static_cast<const void *>(Txt));
-#else
-  std::ignore = Type;
-  std::ignore = Txt;
-#endif
-}
-
 bool Command::enqueue(EnqueueResultT &EnqueueResult, BlockingT Blocking,
                       std::vector<Command *> &ToCleanUp) {
-#ifdef XPTI_ENABLE_INSTRUMENTATION
-  // If command is enqueued from host task thread - it will not have valid
-  // submission code location set. So we set it manually to properly trace
-  // failures if ur level report any.
-  std::unique_ptr<detail::tls_code_loc_t> AsyncCodeLocationPtr;
-  if (xptiTraceEnabled() && !CurrentCodeLocationValid()) {
-    AsyncCodeLocationPtr.reset(
-        new detail::tls_code_loc_t(MSubmissionCodeLocation));
```

Comment:
this class is also used for XPTI only and located in ./sycl/include/sycl/detail/common.hpp, usages could be found in queue ctor, for instance. Although we do not have xpti mentioned in that class declaration & definition so it could be kept useless but present.